site/content/docs/developing/storage.md

```diff
@@ -0,0 +1,219 @@
+
+##### Cleanup
+
+Delete the mount targets using the AWS CLI. 
+
+```bash
+$ aws efs delete-mount-target --mount-target-id $MOUNT_TARGET_1
+$ aws efs delete-mount-target --mount-target-id $MOUNT_TARGET_2
+```
+
+#### With Addons
+Here's an example of how you might create the appropriate EFS infrastructure for an external file system using the [Addons](../developing/additional-aws-resources.md) functionality. 
+
+In a Copilot workspace, create a [Scheduled Job](../manifest/scheduled-job.md) which will never run. We'll use this to deploy our addons template which holds the mount targets we need without worrying about incurring charges for other infrastructure. 
+
+```bash
+$ copilot job init -n efs-job --schedule "cron(0 0 21 10 ? 2015)" -i amazon/amazon-ecs-sample 
+$ copilot job deploy -n efs-job
+```
+
+From the root of your workspace, create the addons directory and a file for the EFS infrastructure to live in. 
+
+```bash
+$ mkdir copilot/efs-job/addons && touch copilot/efs-job/addons/efs.yaml
+```
+Add the following CloudFormation template in `efs.yaml`. 
+
+```yaml
+Parameters:
+  App:
+    Type: String
+    Description: Your application's name.
+  Env:
+    Type: String
+    Description: The environment name your service, job, or workflow is being deployed to.
+  Name:
+    Type: String
+    Description: The name of the service, job, or workflow being deployed.
+
+Resources:
+  EFSFileSystem:
+    Metadata:
+      'aws:copilot:description': 'An EFS File System for persistent backing storage for tasks and services'
+    Type: AWS::EFS::FileSystem
+    Properties:
+      PerformanceMode: generalPurpose 
+      ThroughputMode: bursting
+      Encrypted: true
+
+  MountTargetPublicSubnet1:
+    Type: AWS::EFS::MountTarget
+    Properties: 
+      FileSystemId: !Ref EFSFileSystem
+      SecurityGroups: 
+        - Fn::ImportValue:
+            !Sub "${App}-${Env}-EnvironmentSecurityGroup"
+      SubnetId: !Select 
+        - 0
+        - !Split 
+            - ","
+            - Fn::ImportValue:
+                !Sub "${App}-${Env}-PublicSubnets"
+      
+  MountTargetPublicSubnet2:
+    Type: AWS::EFS::MountTarget
+    Properties:
+      FileSystemId: !Ref EFSFileSystem
+      SecurityGroups:
+        - Fn::ImportValue:
+            !Sub "${App}-${Env}-EnvironmentSecurityGroup"
+      SubnetId: !Select 
+        - 1
+        - !Split 
+            - ","
+            - Fn::ImportValue:
+                !Sub "${App}-${Env}-PublicSubnets"
+
+```
+
+Then, deploy your scheduled job again: 
+```bash
+$ copilot job deploy -n efs-job
+```
+
+This will create an EFS file system and the mount targets needed to allow tasks to attach to it. You can get the ID of this filesystem with the AWS CLI and jq.
+```bash
+$ aws efs describe-file-systems | \
+  jq -r '.FileSystems[] | select((.Tags[]|select(.Key=="copilot-service")|.Value) =="efs-helper") | .FileSystemId'
+```
```

Comment:
nit: instead of doing this can we `Output` the FileSystemID in the stack above so that in the yaml below we can just write:
```yaml
efs:
        id: {{ the FileSystemID output }}
```